checker/src/expression.rs

```diff
@@ -1121,51 +1121,39 @@ impl ExpressionType {
     #[logfn_inputs(TRACE)]
     pub fn is_floating_point_number(&self) -> bool {
         use self::ExpressionType::*;
-        match self {
-            F32 | F64 => true,
-            _ => false,
-        }
+        matches!(self, F32 | F64)
     }
 
     /// Returns true if this type is one of the integer types.
     #[logfn_inputs(TRACE)]
     pub fn is_integer(&self) -> bool {
         use self::ExpressionType::*;
-        match self {
-            I8 | I16 | I32 | I64 | I128 | Isize | U8 | U16 | U32 | U64 | U128 | Usize => true,
-            _ => false,
-        }
+        matches!(
+            self,
+            I8 | I16 | I32 | I64 | I128 | Isize | U8 | U16 | U32 | U64 | U128 | Usize
+        )
     }
 
     /// Returns true if this type is not a primitive type. References are not regarded as
     /// primitives for this purpose.
     #[logfn_inputs(TRACE)]
     pub fn is_primitive(&self) -> bool {
         use self::ExpressionType::*;
-        match self {
-            NonPrimitive | ThinPointer => false, //todo: raw pointer should be primitive
-            _ => true,
-        }
+        !matches!(self, NonPrimitive | ThinPointer)
```

Comment:
The todo comment went away. Is it not an issue any more?

Reply:
It seems a bit stale.